content/blog/2019-09-25-RecordTypes.md

```diff
@@ -0,0 +1,131 @@
+
+
+
+### Instantiation
+To instantiate a new record with a previously declared record type, we use the following format:
+
+    <record instance name>  : <record type> = 
+        record {<field1 name> : <field1 value>; <field2 name> : <field2 value>};
+Where:
+`<record instance name>` is an identifier
+`<record type>` is a previous declared record type
+`<record>` is new keyword
+`<field# name>` is the field name used in the record type definition
+`<field# value>` is an identifier for an existing variable matching the field type
+
+We decided to introduce the `record` keyword to match the precedent of using `id` or `const` in front of an identifier or constant value. Note that the ordering of field name and value pairs do not matter, as long as they match with the definiton’s field names and types. We also only allow field values to be existing variables to match the semantics of current operations. The structure of this statement was designed to match record type declarations as much as possible. 
+
+
+### Access
+We use the dot operator to access a field of a record:
+
+    <new record name> : <type> = <record> . <field>
+
+Where:
+`<new record name>` is a valid identifier that takes on the value of the indicated field
+`<record>` is the name of an instance of a record with a field name `<field>` that has type `<type>`
+Note that there is a space before and after the dot. This is strictly necessary as dot is a valid character for a variable name and we decided that it would be horrible for backwards compatibility if we changed variable naming rules.
+
+
+We chose this format because using the dot operator to access fields is very common in modern programming languages.
+
+### *With* Syntax
+While immutable data structures allow you to more easily reason about how values flow through your program, these currently immutable records are cumbersome to change. 
+For example, to change one field, you must recreate the entire record. While we do want a new record to be created, copying variables between two records is overly tedious. 
+As such, we decided to implement a with syntax, similar to OCaml in which the user just needs to specify a record name as the base, and then the fields desired to be changed, along with the new value. This maintains immutability, without so much tedium.
+
+We use the following syntax:
+
+    <new record>: <record type> = 
+        <old record> with {<field1 name>: <field1 value>; <field2 name>: <field2 value> … };
```

Comment:
As above—you don’t have to invent new syntax here; you could instead try to make it fit as a “normal” instruction.